File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> answer;
    for(int i=0;i<targets.size();i++){
        int res=0;
        for(int j=0;j<targets[i].size();j++){
            int tmp=1000;
            for(int k=0;k<keymap.size();k++){
                for(int l=0;l<keymap[k].size();l++){
                    if(targets[i][j]==keymap[k][l]){
                        tmp=min(tmp, l+1);
                        break;
                    }
                }
            }
            res+=tmp;
        }
        if(res>=1000) res=-1;
        answer.push_back(res);
    }
    return answer;
}

File: 프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;
class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        for(int i=0;i<targets.length;i++){
            int total = 0;
            int cnt = 0;
            for(int j=0;j<targets[i].length();j++){
                char tar = targets[i].charAt(j);
                List<Integer> list = new ArrayList<>();
                for(int k=0;k<keymap.length;k++){
                    for(int l=0;l<keymap[k].length();l++){
                        char cri = keymap[k].charAt(l);
                        if(tar==cri){
                            int turn = l+1;
                            list.add(turn);
                        }
                    }
                }
                Collections.sort(list);
                for(int k=0;k<list.size();k++){
                    total+=list.get(0);
                    cnt++;
                    break;
                }
            }
            if(total==0 || cnt!=targets[i].length()) answer[i]=-1;
            else answer[i]=total;
        }
        return answer;
    }
}